### PR TITLE
Try fix tradebox approval lag

### PR DESCRIPTION
**Pull request by midas-myth**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208378807085384

### CODE REVIEW

File: src/domain/synthetics/tokens/useTokenAllowanceData.ts [hunk 1]
```diff
@@ -51,6 +58,43 @@ export function useTokensAllowanceData(
       }, {} as TokensAllowanceData),
   });
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (gmxer) [inline src/domain/synthetics/tokens/useTokenAllowanceData.ts:61]: no need?

File: src/domain/synthetics/tokens/useTokenAllowanceData.ts [hunk 2]
```diff
@@ -51,6 +58,43 @@ export function useTokensAllowanceData(
...
+  const stableTokenAddresses = useMemo(() => tokenAddresses as Address[], [tokenAddresses.join("-")]);
```
Comment (gmxer) [inline src/domain/synthetics/tokens/useTokenAllowanceData.ts:62]: "stable" means stable coin throughout the project which is confusing, let's use serialized prefix instead?

**Review by gmxer — CHANGES_REQUESTED** [review]

File: src/domain/synthetics/tokens/useTokenAllowanceData.ts (hunk 2 above)
Comment (gdev8317) [reply src/domain/synthetics/tokens/useTokenAllowanceData.ts:62]: trying to understand why is it serialized, it just returns tokenAddresses array?

File: src/domain/synthetics/tokens/useTokenAllowanceData.ts [hunk 3]
```diff
@@ -43,15 +49,58 @@ export function useTokensAllowanceData(
...
+        newData[tokenAddress] = eventValue;
+        continue;
+      }
+
+      const allowance = data.tokenAllowance[tokenAddress];
+      if (eventCreatedAt > data.createdAt) {
+        newData[tokenAddress] = eventValue;
+      } else {
+        newData[tokenAddress] = allowance;
+      }
```
Comment (divhead) [inline src/domain/synthetics/tokens/useTokenAllowanceData.ts:97]: why not simplify it down to something

const event = ...
const multicallData = ...

const dataToApply = [event, multicallData].filter(Boolean).sort(...)[0]

File: src/lib/multicall/useMulticall.ts [hunk 4]
```diff
@@ -163,9 +169,21 @@ export function useMulticall<TConfig extends MulticallRequestConfig<any>, TResul
     },
   });
 
+  const handleMutate: KeyedMutator<TResult | undefined> = useCallback(
+    (...args) => {
+      const opts = args[1];
+      const shouldRevalidate = !(typeof opts !== "boolean" && opts && opts.revalidate === false);
```
Comment (divhead) [inline src/lib/multicall/useMulticall.ts:175]: hard to read this condition
can we simplify it?

Comment (midas-myth) [reply src/lib/multicall/useMulticall.ts]: spread calculation to multiple ifs and added self describing var
```ts
  let shouldRevalidate: boolean | undefined;

      if (opts === false) {
        shouldRevalidate = false;
      } else if (typeof opts !== "boolean" && opts) {
        const isRevalidateExplicitlyDisabled = opts.revalidate === false;
        shouldRevalidate = !isRevalidateExplicitlyDisabled;
      }

```

File: src/domain/synthetics/tokens/useTokenAllowanceData.ts (hunk 3 above)
Comment (midas-myth) [reply src/domain/synthetics/tokens/useTokenAllowanceData.ts:88]: simplified to 

```ts
      const event = approvalStatuses[tokenAddress]?.[spenderAddress];
      const eventValue: bigint | undefined = event?.value;
      const eventCreatedAt: number = event?.createdAt ?? 0;

      const multicallData: bigint | undefined = data?.tokenAllowance[tokenAddress];
      const multicallCreatedAt: number = data?.createdAt ?? 0;

      if (eventCreatedAt > multicallCreatedAt && eventValue !== undefined) {
        newData[tokenAddress] = eventValue;
      } else if (multicallData !== undefined) {
        newData[tokenAddress] = multicallData;
      }
```

**Review by divhead — DISMISSED** [review]

**Review by divhead — APPROVED** [review]

**Review by divhead — APPROVED** [review]

**Review by divhead — DISMISSED** [review]

**Review by divhead — APPROVED** [review]

**Pull request merged**